runtime/recipe/slot-connection.js

```diff
@@ -85,7 +87,10 @@ class SlotConnection {
   }
 
   isResolved() {
-    // TODO: implement
+    if (!this.name || !this.particle || !this.targetSlot)
+      return false;
+    if (this.slotSpec.isRequired && this.targetSlot.sourceConnection == undefined)
```

Comment:
just curios - why do you use
X == undefined here, unlike everywhere else, where it is !X?